tv_apps/browser/js/connection_handler.js

```diff
@@ -0,0 +1,65 @@
+/* global ConnectionManager, ConnectionHandler, Browser */
+'use strict';
+(function(exports) {
+  var ConnectionHandler = function() {};
+
+  ConnectionHandler.prototype = {
+    _active: false,
+    _messageQueue: [],
+
+    init: function() {
+      this._connectionManager = new ConnectionManager();
+      this._connectionManager.init(['iac-webpage-open']);
+      this._connectionManager.on('view', this.preHandleMessage.bind(this));
+    },
+
+    preHandleMessage: function(detail) {
+      if (this._active) {
+        this._digestMessage();
+        this.openPage(detail.url);
+      } else {
+        this._messageQueue.push(detail);
+      }
+    },
+
+    _digestMessage: function() {
+      if (this._messageQueue.length !== 0) {
+        this._messageQueue.forEach(function(detail){
+          this.openPage(detail.url);
+        }.bind(this));
+        this._messageQueue = [];
+      }
+    },
+
+    openPage: function(url) {
+      url = Browser.getUrlFromInput(url);
+      Browser.navigate(url);
```

Comment:
We probably should check whether the current tab is empty or not. If it's not, maybe we have to open the url in a new tab? Otherwise only the last queued URL in ```this._messageQueue``` will be opened.